File: programmers/문자열 내 마음대로 정렬하기.py
def solution(strings, n):
    answer = []
    array = [[] for _ in range(len(strings))]
    for idx, word in enumerate(strings):
        array[idx].append(word)

    array = sorted(array, key=lambda x: (x[0][n], x[0]))
    for string in array:
        answer.append(string[0])
    return answer

File: programmers_level1/문자열_내_마음대로_정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int idx; // comparator에 비교하고자 하는 두 대상 외에 다른 parameter 들어가지 못하므로 전역 변수 선언

bool compareByIdx(string &a, string &b) {
    if (a[idx] == b[idx]) {
        return a < b;
    }
    return a[idx] < b[idx];
}

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer(strings);
    idx = n;
    sort(answer.begin(), answer.end(), compareByIdx);
    return answer;
}